### ISSUE TITLE
git log --all

### ISSUE THREAD

**Issue by michaelAtCoalesce** [isomorphic-git/isomorphic-git#1449]
i know there is already a git log that allows you to get history from a single commit - but we would like to be able to do the equivalent of a git log --graph --all, where all commits from all branches are returned in a single array. 

this is so we can do some kind of a visualization like this 
![image](https://user-images.githubusercontent.com/96628583/147836241-58224f79-b186-4ea1-ba89-84525a578aef.png)

so this would be an enhancement on the current git log functionality to allow returning of all commits from all branches, not just one

**Comment by jcubic** [comment]
I think that you already have everything to implement this. The code may be complex since you need to connect the commits into a graph, but the library will need to do the same.

Unfortunately no one will create the code for you. This project is not actively developed by the original author, so the only way to have this feature is to make it yourself and contribute or wait (possible forever) for someone that will code this for you, because he will need the same feature.

**Comment by michaelAtCoalesce** [comment]
yeah i am writing the code as we speak - 

how do i go from branch name as a string to the oid it points to? when i use resolveRef and pass in the branch name it says it isnt found

**Comment by jcubic** [comment]
It works for me: https://isomorphic-git.org/docs/en/resolveRef

**Comment by michaelAtCoalesce** [comment]
yeah looks like i was using listbranches with remote 'origin'. i need to understand where we are only cloning only down one branch. thanks

**Comment by michaelAtCoalesce** [comment]
I’ll take a look at the source and see how things work..

Iterating through the branches is quite slow especially because it has to
repeat the commits over and over again

Will report back when i figure things out more
